base/libgit2/utils.jl

```diff
@@ -37,12 +37,16 @@ function prompt(msg::AbstractString; default::AbstractString="", password::Bool=
     end
     msg = !isempty(default) ? msg*" [$default]:" : msg*":"
     uinput = if password
-        Base.getpass(msg)
+        Base.getpass(msg)  # Automatically chomps. We cannot tell EOF from '\n'.
     else
         print(msg)
-        readline()
+        readline(chomp=false)
     end
-    isempty(uinput) ? default : uinput
+    if !password
+        isempty(uinput) && return Nullable{String}()  # Encountered EOF
+        uinput = chomp(uinput)
+    end
+    Nullable{String}(isempty(uinput) ? default : uinput)
```

Comment:
Good catch. This is used by [PkgDev](https://github.com/JuliaLang/PkgDev.jl/blob/0f8f2b4c3a68572e178c5e479a0d43b826b2d5e6/src/PkgDev.jl#L113). 

What's the recommended procedure for deprecating a function's return type? Should I deprecate `prompt(::AbstractString)` and introduce the following:

```julia
prompt(::Type{Nullable{String}}, msg::AbstractString; default::AbstractString="", password::Bool=false)
```